modules/sqs-queue/main.tf

```diff
@@ -69,3 +70,13 @@ resource "aws_sqs_queue_policy" "sqs_queue_policy" {
   queue_url = module.sqs_queue.url
   policy    = one(module.queue_policy[*].json)
 }
+
+resource "aws_sqs_queue_redrive_policy" "dead_letter_queue" {
+  count = local.redrive_policy_enabled != null ? 1 : 0
+
+  queue_url = module.sqs_queue.url
+  redrive_policy = jsonencode({
+    deadLetterTargetArn = var.dead_letter_sqs_url != null ? var.dead_letter_sqs_url : one(module.dead_letter_sqs_remote_state[*].outputs.arn)
```

Comment:
Just a sanity check here... This looks like you're assigning a URL to an ARN. Don't you need to look up the sqs queue with [a data query](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/sqs_queue) if you want the ARN? Otherwise, I'm not sure this would work.

Reply:
Misnamed the variable, that should've been an arn.